sklearn/tests/test_isotonic.py

```diff
@@ -27,6 +27,14 @@ def test_permutation_invariance():
     assert_array_equal(y_transformed, y_transformed_s)
 
 
+def test_check_increasing_small_number_of_samples():
+    x = [0, 1, 2]
+    y = [1, 1.1, 1.05]
+
+    is_increasing = assert_no_warnings(check_increasing, x, y)
+    assert_true(is_increasing)
```

Comment:
I think it is correct. Fitting a straight line to these 3 data points would also result in a positive slope. The correlation coefficient is 0.5, so positive. I plotted the data for illustrating it:
![screen shot 2017-03-21 at 14 48 59](https://cloud.githubusercontent.com/assets/23242456/24150265/8bdd1ff4-0e45-11e7-8848-f6b92317749a.png)

This check is based on the correlation coeffcient and not the difference between the values (np.all(np.diff([1, 1.1, 1.05]) > 0) is here indeed false)